### PR TITLE
feat(zone.js): Provide auto fakeAsync() functionality

### PR DESCRIPTION
**Pull request by JiaLiPassion**
`Zone.js` provides an integration with the `jasmine.clock()`,

```
describe(`Auto fakeAsync() with jasmine.clock()`, () => {
  beforeAll(() => jasmine.clock().install());
  afterAll(() => jasmine.clock().uninstall());
  it('test1', () => { // The test becomes fakeAsync() automatically, don't need to call fakeAsync()
    let t = 0;
    setTimeout(() => {
      t = t + 1;
    }, 100);
    expect(t).toBe(0);
    tick(100);
    expect(t).toBe(1);
  });
});
```

This PR provides two new APIs to make the functionality available not only with `jasmine.clock()`.

```
describe('Auto fakeAsync()', () => {
  beforeAll(() => enableAutoFakeAsync());
  afterAll(() => disableAutoFakeAsync());

  it('should auto become fakeAsync() test', () => {
    let t = 0;
    setTimeout(() => {
      t = t + 1;
    }, 100);
    expect(t).toBe(0);
    tick(100);
    expect(t).toBe(1);
  });
});
```

This PR is also related to https://github.com/angular/angular/pull/40611.

### CODE REVIEW

File: aio/content/guide/testing-components-scenarios.md [hunk 1]
```diff
@@ -461,6 +461,17 @@ import 'zone.js/dist/zone-testing';
   region="fake-async-test-clock">
 </code-example>
 
+#### Automatically become fakeAsync() test
+
+When using `jasmine.clock()`, the test case becomes `fakeAsync()` automatically, the same
```
Comment (jessicajaniuk) [inline aio/content/guide/testing-components-scenarios.md:466]:
```suggestion
When using `jasmine.clock()`, the test case becomes `fakeAsync()` automatically. The same
```

File: packages/core/testing/src/fake_async.ts [hunk 2]
```diff
@@ -152,3 +152,40 @@ export function flushMicrotasks(): void {
   }
   throw new Error(fakeAsyncTestModuleNotLoadedErrorMessage);
 }
+
+/**
+ * Enable automatically go into the `fakeAsync()` test.
+ *
+ * describe('Auto go into fakeAsync()', () => {
```
Comment (jessicajaniuk) [inline packages/core/testing/src/fake_async.ts:159]: Maybe prepend this with "Example:". I was confused why a test was fully commented out here for a second.

**Review by jessicajaniuk — APPROVED** [review]
LGTM 🍪 

Just some minor language suggestions.

**Review by mhevery — CHANGES_REQUESTED** [review]
Perhaps I am misunderstanding this PR, so please correct me.  To me it seems like jasmine.clock() turns on `fakeAsync` functionality. The issue is that `fakeAsync` does a lot more than just work with clock. For example `Promise`s are now also async. I don't think this kind of magic is a good idea as adding `jasmine.clock()` changes a lot more than just make `setTimeout` into mockable version.

**Comment by JiaLiPassion** [comment]
@mhevery , yeah, so the motivation of this PR is just make a existing feature of zone.js easier to use.
When the user is using `jasmine.clock()`, it will turn on `fakeAsync()` automatically, the user doesn't need to write the code `fakeAsync()` and the test cases automatically becomes `fakeAsync()` test, so I just think if we make this feature also available without using `jasmine.clock()`.

**Comment by mhevery** [comment]
> The benefit is just let user write a little bit less code.

I think this is too much magic. I think we should stick to explicit `fakeAsync`. We can discuss this on our team meeting.

**Comment by JiaLiPassion** [comment]
@mhevery, got it, I agree explicit `fakeAsync` is better to understand, and I just have this idea when I am working on https://github.com/angular/angular/pull/40611, since when the user is using `jest` or `jasmine.clock`, they don't need to call some explicit function such as `fakeAsync()`, so I think maybe in `zone.js` we can provide the similar functionality.

**Pull request closed without merging**